app/main.py

```diff
@@ -1,3 +1,92 @@
+from __future__ import annotations
+
+
 class Distance:
-    # Write your code here
-    pass
+
+    def __init__(self, km: float) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: Distance | float) -> Distance:
+        if isinstance(other, (float, int)):
+            return Distance(self.km + other)
+        elif isinstance(other, Distance):
+            return Distance(self.km + other.km)
+        else:
+            raise TypeError(f"Unsupported operation '+'"
+                            f" ({type(self)}) and ({type(other)})")
```

Comment:
u don't need to raise TypeError
also remove redundant elif check
fix it in all places